javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java

```diff
@@ -384,12 +392,21 @@ private void matchTypeParameters(ResolvedType expectedType, ResolvedType actualT
     }
 
     private Optional<MethodUsage> solveMethodAsUsage(ResolvedTypeVariable tp, String name, List<ResolvedType> argumentsTypes, Context invokationContext) {
-        for (ResolvedTypeParameterDeclaration.Bound bound : tp.asTypeParameter().getBounds()) {
+        List<ResolvedTypeParameterDeclaration.Bound> bounds = tp.asTypeParameter().getBounds();
+
+        if (bounds.isEmpty()) {
```

Comment:
Ditto here

My preference would be to add a line comment here with a short note about why we're doing this and to include a link to the GitHub issue. This will help any future readers of the code :)